File: examples/coulomb/coulomb.cxx
#include <iostream>

#include "Eisvogel/Common.hh"
#include "Eisvogel/SignalCalculatorOld.hh"
#include "Eisvogel/SparseCurrentDensity3D.hh"
#include "Eisvogel/SignalExport.hh"
#include "Eisvogel/CoordUtils.hh"

namespace CU = CoordUtils;

int main(int argc, char* argv[]) {

  if(argc < 2) {
    throw;
  }

  std::string wf_path = argv[1];
  SignalCalculatorOld calc(wf_path);

  DeltaVector voxel_size = CU::MakeCoordVectorTXYZ(0.1, 0.1, 0.1, 0.1);

  // test trajectory: a point charge moving parallel to the x-axis 
  // with a constant impact parameter of 'b' along the z-axis
  scalar_t b = 10;
  scalar_t tstart = -250, tend = 250;
  scalar_t charge = 1;
  scalar_t beta = 0.9;

  SparseCurrentDensity3D curr_dist(voxel_size);

  for(scalar_t cur_t = tstart; cur_t <= tend; cur_t += CU::getT(voxel_size)) {

    scalar_t cur_x = beta * cur_t;
    scalar_t cur_y = 0;
    scalar_t cur_z = b;
    
    scalar_t cur_jx = charge * beta / curr_dist.getVolumeElementXYZ();
    scalar_t cur_jy = 0;
    scalar_t cur_jz = 0;

    curr_dist.addCurrentElement({
	CU::MakeCoordVectorTXYZ(cur_t, cur_x, cur_y, cur_z),
	CU::MakeFieldVectorXYZ(cur_jx, cur_jy, cur_jz)
    });
  }

  std::cout << "Computing signal ..." << std::endl;
  std::vector<scalar_t> signal_times, signal_values;
  for(scalar_t cur_t = -10; cur_t < 20; cur_t += 1) {
    scalar_t cur_signal = calc.ComputeSignal(curr_dist, cur_t);
    signal_times.push_back(cur_t);
    signal_values.push_back(cur_signal);
  }

  ExportSignal(signal_times, signal_values, "./test_signal.csv");

  return 0;
}

File: examples/coulomb/coulomb.py
import argparse
import numpy as np
from eisvogel import DeltaVector, CoordVector, FieldVector, SignalCalculator, SparseCurrentDensity3D

def compute_signal(wf_path):

    b = 10
    tstart = -250.0
    tend = 250.0
    charge = 1
    beta = 0.9

    voxel_size = DeltaVector.FromDeltaTXYZ(0.1, 0.1, 0.1, 0.1)
    curr_dist = SparseCurrentDensity3D(voxel_size)

    for cur_t in np.linspace(tstart, tend, 5000):
        
        cur_x = beta * cur_t
        cur_y = 0
        cur_z = b

        cur_jx = charge * beta / (0.1 ** 3)
        cur_jy = 0
        cur_jz = 0

        curr_dist.addCurrentElement(CoordVector.FromTXYZ(cur_t, cur_x, cur_y, cur_z),
                                    FieldVector.FromXYZ(cur_jx, cur_jy, cur_jz))

    calc = SignalCalculator(wf_path)

    for cur_t in range(-10, 20):
        cur_sig = calc.ComputeSignal(curr_dist, cur_t)
        print(f"{cur_t}: {cur_sig}")

if __name__ == "__main__":

    parser = argparse.ArgumentParser()
    parser.add_argument("--wf_path", action = "store", dest = "wf_path")
    args = vars(parser.parse_args())
    
    compute_signal(**args)
